addons/udes_stock_routing/models/stock_quant.py

```diff
@@ -0,0 +1,49 @@
+from odoo import models, api
+from .stock_rule import RULE_RESERVATION_TYPE_WHOLE_PALLET
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    def _gather(self, product_id, location_id, **kwargs):
+        """
+        Extend _gather to filter by split pick rules, which were passed in via context.
+        """
+        StockQuant = self.env["stock.quant"]
+
+        split_pick_rule = self.env.context.get("split_pick_rule", None)
```

Comment:
Same thing in fact, can resolve this just if you like more
```suggestion
        split_pick_rule = self._context.get("split_pick_rule", None)
```